source/frontend/src/features/documents/list/DocumentResults/DocumentResultsColumns.tsx

```diff
@@ -112,46 +132,103 @@ const renderActions = (
 ) => {
   return function ({ row: { original, index } }: CellProps<DocumentRow, string>) {
     const { hasClaim } = useKeycloakWrapper();
-    return (
-      <StyledIconsRow className="no-gutters">
-        {hasClaim(Claims.DOCUMENT_VIEW) && (
-          <Col>
-            <Button
-              data-testid="document-view-button"
-              icon={<FaEye size={24} id={`document-view-${index}`} title="document view details" />}
-              onClick={() => original?.id && onViewDetails(DocumentRow.toApi(original))}
-            ></Button>
-          </Col>
-        )}
-        {hasClaim(Claims.DOCUMENT_DELETE) && (
-          <Col>
+
+    const documentInError =
+      (original.mayanDocumentId === null || original.mayanDocumentId === -1) &&
```

Comment:
This logic appears inconsistent across this file. Also, I believe you can use isValidId to be consistent with the rest of our code base.

Reply:
In my testing, mayanId is being converted from null to 0 on the backend, so none of these icons are being displayed.

Reply:
Also, it looks like some of these boolean statements can be collapsed to make them more readable (ie: original.queueStatusTypeCode && original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.PENDING
is equivalent to original?.queueStatusTypeCode === ApiGen_CodeTypes_DocumentQueueStatusTypes.PENDING

Reply:
Updated